aten/src/ATen/native/Bucketization.cpp

```diff
@@ -105,7 +112,10 @@ Tensor& searchsorted_out_cpu(const Tensor& sorted_sequence, const Tensor& self,
   searchsorted_maybe_trim_input_tensors(trimmed_input, trimmed_boundaries, self, sorted_sequence);
   const Tensor& final_input = trimmed_input.defined() ? trimmed_input : self;
   const Tensor& final_boundaries = trimmed_boundaries.defined() ? trimmed_boundaries : sorted_sequence;
-  dispatch(result, final_input, final_boundaries, out_int32, right);
+  dispatch(out, final_input, final_boundaries, out_int32, right);
+  if (!out.is_contiguous()) {
+    result.copy_(out);
+  }
   return result;
```

Comment:
Can you combine lines 104-107 above with these ones so there's a single return statement. The two different dispatch calls can be in an if-else statement.